### PR TITLE
Mes 2601 implement template id provider

### PR DESCRIPTION
**Pull request by szabikr**
- Implementing TemplateIdProvider
- Using it wherever it's needed
- Making sure that unit tests run

### CODE REVIEW

File: src/functions/sendCandidateResults/application/service/template-id-provider.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+export function isFail(activityCode: ActivityCode) : boolean {
+  return activityCode === '2' || activityCode === '3';
+}
+
+export function isTerminated(activityCode: ActivityCode) : void {
+  throw new DocumentsServiceError(
+     0 ,
+     `Failed to get template id for activity code { activityCode }`,
```
Comment (MarkPatt89) [inline src/functions/sendCandidateResults/application/service/template-id-provider.ts]: Will this still print output without '$'?

**Review by MarkPatt89 — APPROVED** [review]
Just one question on the missing $.

**Review by TomBillingtonUK — APPROVED** [review]
Nice One :)

File: src/functions/sendCandidateResults/application/service/template-id-provider.ts (hunk 1 above)
Comment (TomBillingtonUK) [reply src/functions/sendCandidateResults/application/service/template-id-provider.ts]: Fixed

**Pull request merged**